src/server.rs

```diff
@@ -298,7 +299,7 @@ pub type BoxHandler = Box<dyn CloneableHandler>;
 #[derive(Clone)]
 pub struct RequestCallContext {
     server: Arc<ServerCore>,
-    registry: Arc<UnsafeCell<HashMap<&'static [u8], BoxHandler>>>,
+    registry: Rc<UnsafeCell<HashMap<&'static [u8], BoxHandler>>>,
```

Comment:
Because this struct can be send to different threads, using `Rc` is not safe.

Reply:
`UnsafeCell<HashMap<&[u8], Box<dyn CloneableHandler>>>` is not `Sync`, so `Arc<UnsafeCell<HashMap<&'static [u8], BoxHandler>>>` is neither `Send` nor `Sync`, and is not safe to send between threads. The only place that `registry` is used is by `RequestCallContext::get_handler`, which is unsafe and notes that it must always be called from the same thread.

If this safety requirement is not being met then the code is already unsound. If it is being met, then using an `Rc` rather than an `Arc` is fine.

Reply:
If you actually want to share it between threads, then this should use a `Mutex` rather than an `UnsafeCell`.

Reply:
Sent #645 to do this.

Reply:
> is neither Send nor Sync, and is not safe to send between threads.

It's true generally speaking. But it's safe here because each request call will be bound to a thread and never be accessed from different threads at the same time. Using `Arc` instead of `Rc` because it can still be sent around, hence may be clone and drop from different threads at the same time. For example,

thread 1 | thread 2
--|--
clone    |
send -> |
|            | access handler
 drop       | drop